### PR TITLE
Convert  ServicesView and ODR Card AB#15795

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#15795](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15795)

## Description

1. Convert ServicesView component
2. Convert ODR Card Component
3. Adjust patientDataStore to expose variables that can be susbscribed to.
4. Add gridUtilities to assist in determining column breakpoints.

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

### Card:
image

### Sensitive Download
image

### Tooltip of details
image

### Downloading processing
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            )
+            .then((blob) => saveAs(blob, `Organ_Donor_Registration.pdf`))
+            .catch((err) => logger.error(err));
+    }
+}
+
+function showConfirmationModal(): void {
+    sensitiveDocumentModal.value?.showModal();
+}
+</script>
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:77]: There used to be code here that added a custom error when this service was blocked. Is that handled differently now?

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        <template #icon>
+            <img
+                src="@/assets/images/services/odr-logo.svg"
+                alt="Organ Donor Registry Logo"
+                :height="30"
+            />
+        </template>
+        <div class="flex-grow-1 d-flex flex-column text-body-1 card-content">
+            <div>
+                <span class="text-grey-darken-2">Status: </span>
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:93]: (Applies throughout.)

Suggest using `text-medium-emphasis` class instead, or none at all.

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                }}</strong>
+                <v-tooltip
+                    data-testid="organ-donor-registration-status-info-popover"
+                    location="top start"
+                    content-class="bg-grey-darken-3"
+                    :text="registrationData?.statusMessage"
+                >
+                    <template v-slot:activator="{ props }">
+                        <v-icon
+                            ata-testid="organ-donor-registration-status-info-button"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:105]:
```suggestion
                            data-testid="organ-donor-registration-status-info-button"
```

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                    />
+                    <v-icon
+                        v-else
+                        icon="download"
+                        size="medium"
+                        square
+                        aria-hidden="true"
+                    />
+                    <span class="text-body-1 ml-3">Download</span>
+                </HgButtonComponent>
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:137]: Vuetify has a loading prop available on the button component we could use instead.

```suggestion
                <HgButtonComponent
                    v-if="registrationData?.registrationFileId"
                    data-testid="organ-donor-registration-download-button"
                    variant="secondary"
                    :prepend-icon="download"
                    :text="Download"
                    :loading="isLoadingFile"
                    @click="showConfirmationModal"
                />
```

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                    data-testid="organ-donor-registration-decision-no-file"
+                    >Not Available</strong
+                >
+            </div>
+            <div class="text-body-1">
+                <p>
+                    <a
+                        href="http://www.transplant.bc.ca/Pages/Register-your-Decision.aspx"
+                        target="_blank"
+                        data-testid="organ-donor-registration-link"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:149]:
```suggestion
                        class="text-link"
                        data-testid="organ-donor-registration-link"
```

File: Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue:98]: I see that code was from a recent change in the dev branch.  I've now ported those changes to this branch.

**Review by MikeLyttle — APPROVED** [review]

**Comment by BronzBierd** [comment]
@MikeLyttle  Thanks for picking this up see you on Thursday.

**Pull request merged**